### PR TITLE
Add scenes2strip version to strip DEM metadata handling

### PR DESCRIPTION
**Pull request by clairecporter**

### CODE REVIEW

File: lib/dem.py [hunk 1]
```diff
@@ -107,6 +107,7 @@
                                    dem.tif\Z""", re.I| re.X)
 
 xtrack_sensor_pattern = re.compile("[wqg]\d[wqg]\d", re.I)
+s2s_version_pattern = re.compile("Strip Metadata( \(v(?P<s2sversion>\d[\d\.]*)\))?")
```
Comment (ehusby) [inline lib/dem.py:110]: I think the regex is off on the version number parsing, should instead be `"Strip Metadata( \(v(?P<s2sversion>\d+[\.\d+]*)\))?"`

File: lib/dem.py [hunk 2]
```diff
@@ -890,9 +903,12 @@ def get_metafile_info(self):
             try:
                 self.avg_acqtime1 = datetime.strptime(metad['STRIP_DEM_avgAcqTime1'], "%Y-%m-%d %H:%M:%S")
                 self.avg_acqtime2 = datetime.strptime(metad['STRIP_DEM_avgAcqTime2'], "%Y-%m-%d %H:%M:%S")
-            except KeyError as e:
-                self.avg_acqtime1 = datetime.strptime(metad['STRIP_DEM_avgAcqTime'], "%Y-%m-%d %H:%M:%S")
-                self.avg_acqtime2 = self.avg_acqtime1
+            except KeyError:
+                try:
+                    self.avg_acqtime1 = datetime.strptime(metad['STRIP_DEM_avgAcqTime'], "%Y-%m-%d %H:%M:%S")
+                    self.avg_acqtime2 = self.avg_acqtime1
```
Comment (ehusby) [inline lib/dem.py:909]: @clairecporter Do you foresee the possibility of there being a single 'STRIP_DEM_avgAcqTime' field in the mdf file? That somewhat breaks the concept of the separate average acquisition times, so I don't think we should handle it at all?

File: index_setsm.py [hunk 3]
```diff
@@ -648,8 +648,8 @@ def write_to_ogr_dataset(ogr_driver_str, ogrDriver, dst_ds, dst_lyr, groups, pai
                                 'SENSOR2': record.sensor2,
                                 'ACQDATE1': record.acqdate1.strftime('%Y-%m-%d'),
                                 'ACQDATE2': record.acqdate2.strftime('%Y-%m-%d'),
-                                'AVGACQTM1': record.avg_acqtime1.strftime("%Y-%m-%d %H:%M:%S"),
-                                'AVGACQTM2': record.avg_acqtime2.strftime("%Y-%m-%d %H:%M:%S"),
+                                'AVGACQTM1': record.avg_acqtime1.strftime("%Y-%m-%d %H:%M:%S") if record.avg_acqtime1 else None,
+                                'AVGACQTM2': record.avg_acqtime2.strftime("%Y-%m-%d %H:%M:%S") if record.avg_acqtime2 else None,
```
Comment (ehusby) [inline index_setsm.py:652]: @clairecporter I think we can get these field values rebuilt when indexing from JSONs. Let's discuss offline

**Review by ehusby — CHANGES_REQUESTED** [review]

File: lib/dem.py (hunk 1 above)
Comment (clairecporter) [reply lib/dem.py:110]: They look to be the same thing to me.  In both cases, you need a digit followed by nothing or any combination of digits and periods.

File: lib/dem.py (hunk 2 above)
Comment (clairecporter) [reply lib/dem.py:909]: No, but you wrote those lines...  I'm happy to take it out.

File: lib/dem.py (hunk 1 above)
Comment (ehusby) [reply lib/dem.py:110]: Ah you're right, I forgot that the characters in the square brackets can be in any order

File: index_setsm.py (hunk 3 above)
Comment (clairecporter) [reply index_setsm.py:652]: This fix was for if they are not in the MDF file from an old MDF calculated before these changes.  I don't see it happening, but I don't want the code to break if it does.

File: lib/dem.py (hunk 1 above)
Comment (ehusby) [reply lib/dem.py:110]: No need to change. Actually I don't know how regex would handle a `\d+` in the square brackets...

**Review by ehusby — APPROVED** [review]

**Comment by clairecporter** [comment]
@ehusby , tests updated.  I will merge if you are happy.

**Comment by ehusby** [comment]
I don't have a "happy" emoji to react with, but anways, you already merged! All good.

**Pull request merged**